### DISCUSSION TITLE
contract deployment via smart wallet

### DISCUSSION THREAD

**Discussion by dckc** [Agoric/agoric-sdk#7316] [Ideas]
@samsiegart just suggested this. I think it has come up in office hours before with folks like @anilhelvaci...

The smart wallet handles `E(zoe).offer(...)` for you. It could do `E(zoe).install(...)` and `E(zoe).startInstance(...)` too.

It would let you keep things like `creatorFacet` in your smart wallet. Maybe something like the `callPipe` pattern could be used to call methods on such things.

cc @turadg

**Comment by samsiegart** [comment]
Thanks for starting this.

I can't remember if `creatorInvitation` is preferable and `creatorFacet` is going out of style, but if so that would already work with what we have. The creator could accept an invitation that returns a bunch of invitation makers in the offer result, much like how the EC works.

**Comment by dckc** [comment]
my impression is the other way around: `creatorInvitation` is going out of style. _something something_ durability _something_.

**Comment by carlos-kryha** [comment]
Would you then use a custom frontend application to handle the deployment (over a local bridge) or would it be part of the smar-wallet's ui? Maybe some other way?

**Reply by samsiegart** [reply]
That sounds feasible, you can also interact with the smart wallet through a CLI.

**Reply by carlos-kryha** [reply]
I'm a bit confused about the naming, but for these comments let me just refer to the smart-wallet contracts as `smart-wallet` and the smart-wallet ui (https://wallet.agoric.app/wallet/) as `web-wallet`. 
Is it currently possible to interact with the `web-wallet` through a CLI?

**Reply by turadg** [reply]
> refer to the smart-wallet contracts as `smart-wallet`

👍  though a rename is coming: https://github.com/Agoric/agoric-sdk/issues/7216

> the smart-wallet ui (https://wallet.agoric.app/wallet/) as `web-wallet`

The implementation of what's served at wallet.agoric.app is https://github.com/Agoric/wallet-app/ so we call it `wallet-app`. 

It is not _the_ UI to the smart-wallet contract because there can be many. Multiple command-line UIs in https://github.com/Agoric/agoric-sdk/tree/master/packages/agoric-cli can send/receive messages to/from the smart-wallet/account contract. 

> Is it currently possible to interact with the web-wallet through a CLI?

Yes, check out https://github.com/Agoric/agoric-sdk/blob/master/packages/agoric-cli/test/agops-perf-smoketest.sh

And the CLI usability is improving in https://github.com/Agoric/agoric-sdk/pull/7256

**Reply by carlos-kryha** [reply]
Great! That CLI functionality will definitely come in handy for testing while we figure out the wallet-app integration, I'll check it out, thank you

**Comment by dckc** [comment]
Another option, that doesn't require changing the walletFactory contract: [contract-starter.js](https://gist.github.com/dckc/3832d8c5739cb933f6dbe906fbfb3fa7#file-contract-starter-js)

I'm playing around with it this morning; started a [contract starter contract](https://discord.com/channels/585576150827532298/1183019528637128784) thread in discord `#dev`.

cc @turadg

**Reply by dckc** [reply]
broke ground:

2023-12-09 11:22 5846a31366 test: terminate a contract by upgrading it to a trivial impl (WIP)
https://github.com/Agoric/agoric-sdk/pull/8640

got the postalSvc contract started in a bootstrap test

p.s. https://github.com/Agoric/agoric-sdk/pull/8640

**Reply by dckc** [reply]
contractStarter work has moved to ag-power-tools. For example, see https://github.com/agoric-labs/ag-power-tools/pull/7#issuecomment-1891389527
